src/components/filter-modal/filter-modal.js

```diff
@@ -0,0 +1,136 @@
+
+export const FilterModal = ({ placeholder, values, isOpen, selectValue }) => {
+  const [selectedValue, setSelectedValue] = useState("");
+
+  const onValueChangedHandler = value => {
+    setSelectedValue(value);
+  };
+
+  const onValueSelectedHandler = value => {
+    if (selectValue) {
+      selectValue(value);
+    }
+  };
+
+  if (!isOpen) {
+    return null;
+  }
+
+  const WithScrollDiv = WithScrollHandler(({ children, ...props }) =>
+    React.createElement("div", props, children)
+  );
+
+  return (
+    <div className="filter-modal__container">
+      <div className="filter-modal">
+        <SearchInput
+          hasIcon
+          value={selectedValue}
+          placeholder={placeholder}
+          onValueChange={onValueChangedHandler}
+        />
+        <div className="filter-modal__separator" />
+        <div className="filter-modal__scrollable-wrapper">
+          <WithScrollDiv className="filter-modal__list">
+            {(hasReachedTop, hasReachedBottom) => (
+              <React.Fragment>
+                {values
+                  .filter(
+                    value => !selectedValue || value.includes(selectedValue)
+                  )
+                  .map(value => (
+                    <div
+                      key={`FilterModalValue${value}`}
+                      className="filter-modal__value"
+                      onClick={() => onValueSelectedHandler(value)}
+                    >
+                      {value}
+                    </div>
+                  ))}
+                <div className="filter-modal__scrollarea">
+                  <span className="icon">
+                    {!hasReachedTop && <ChevronUpIcon />}
+                  </span>
+                  <span className="icon">
+                    {!hasReachedBottom && <ChevronDownIcon />}
+                  </span>
+                </div>
+              </React.Fragment>
+            )}
+          </WithScrollDiv>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+FilterModal.defaultProps = {
+  isOpen: false,
+  values: [],
+  placeholder: "cauta"
+};
+
+FilterModal.propTypes = {
+  isOpen: PropTypes.bool,
+  placeholder: PropTypes.string,
+  values: PropTypes.array,
+  selectValue: PropTypes.func
+};
+
+const WithScrollHandler = WrappedComponent => {
+  const WrapperComponent = ({ children, ...rest }) => {
+    const [contentHeight, setContentHeight] = useState(0);
+    const [hasReachedTop, setHasReachedTop] = useState(true);
+    const [hasReachedBottom, setHasReachedBottom] = useState(false);
+
+    const onScrollHandler = event => {
+      let updatedContentHeight = contentHeight ? contentHeight : 0;
+
+      if (!updatedContentHeight) {
+        event.target.children.forEach(child => {
+          if (child.className.includes("filter-modal__value")) {
+            updatedContentHeight += child.offsetHeight;
+          }
+        });
+
+        setContentHeight(updatedContentHeight);
+      }
```

Comment:
I understand it can get quite expensive for a large list, so an optimization could be debouncing the scroll handler.
However, I am not so sure about implementing a custom scrollbar with cross-browser support.
According to the designs, should there be only a normal scrollbar or with these arrows?

Reply:
> I understand it can get quite expensive for a large list, so an optimization could be debouncing the scroll handler.

Unless we need that custom scrollbar, I think we can easily use CSS for scrolling.

> However, I am not so sure about implementing a custom scrollbar with cross-browser support.

Definitely agree. This is a no-go. I don't think it's worth.

> According to the designs, should there be only a normal scrollbar or with these arrows?

We need to ask the design team about this one. Will try to reach out to them and come back with an update.

Reply:
For this version is alright to go ahead with native scrolling. Lets keep it simple 🙌